This patch is from upstream gcc-rs pull request #151 (authored by yours
truly). Its purpose is to be less strict on the vendor part of the target
triple with respect to detecting armv7 support. While treating target triples
this way is far from ideal, it is pervasive and this patch should do for our
cases.

See: https://github.com/alexcrichton/gcc-rs/pull/151

diff --git a/src/vendor/gcc/src/lib.rs b/src/vendor/gcc/src/lib.rs
index 43cc371..934f677 100644
--- a/src/vendor/gcc/src/lib.rs
+++ b/src/vendor/gcc/src/lib.rs
@@ -515,7 +515,7 @@ impl Config {
             }
 
             // armv7 targets get to use armv7 instructions
-            if target.starts_with("armv7-unknown-linux-") {
+            if target.starts_with("armv7-") && target.contains("-linux-") {
                 cmd.args.push("-march=armv7-a".into());
             }
